### PR TITLE
Custodial Assets 

### PR DESCRIPTION
**Pull request by giskook**
Closes #<issue number>.

### What does this PR do?

Custodial Assets

### Reviewers

Main reviewers:

- @John
- @Doe

Codeowner reviewers:

- @-Alice
- @-Bob

### CODE REVIEW

File: ethtxmanager/custodialassets.go [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+	err = json.Unmarshal(retBytes, &args)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal tx %x data: %v", tx.Hash(), err)
+	}
+
+	return &args, nil
+}
+
+func (c *Client) unpackVerifyBatchesTrustedAggregatorTx(tx *types.Transaction) (*verifyBatchesTrustedAggregatorArgs, error) {
+	if tx == nil || len(tx.Data()) == 0 {
```
Comment (zjg555543) [inline ethtxmanager/custodialassets.go:124]: len must large then sigLen

File: ethtxmanager/custodialassets.go [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
..."}],"stateMutability":"view","type":"function"},{"inputs":[{"internalType":"uint64","name":"pendingStateNum","type":"uint64"},{"internalType":"uint64","name":"initNumBatch","type":"uint64"},{"internalType":"uint64","name":"finalNewBatch","type":"uint64"},{"internalType":"bytes32","name":"newLocalExitRoot","type":"bytes32"},{"internalType":"bytes32","name":"newStateRoot","type":"bytes32"},{"internalType":"bytes32[24]","name":"proof","type":"bytes32[24]"}],"name":"verifyBatches","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[{"internalType":"uint64","name":"pendingStateNum","type":"uint64"},{"internalType":"uint64","name":"initNumBatch","type":"uint64"},{"internalType":"uint64","name":"finalNewBatch","type":"uint64"},{"internalType":"bytes32","name":"newLocalExitRoot","type":"bytes32"},{"internalType":"bytes32","name":"newStateRoot","type":"bytes32"},{"internalType":"bytes32[24]","name":"proof","type":"bytes32[24]"}],"name":"verifyBatchesTrustedAggregator","outputs":[],"stateMutability":"nonpayable","type":"function"},{"inputs":[],"name":"version","outputs":[{"internalType":"uint256","name":"","type":"uint256"}],"stateMutability":"view","type":"function"}]`
```
[the remark this answers is not in the document]
Comment (giskook) [reply ethtxmanager/custodialassets.go]: done

File: ethtxmanager/custodialassetshttp.go [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+
+	"github.com/0xPolygonHermez/zkevm-node/hex"
+	"github.com/0xPolygonHermez/zkevm-node/log"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/google/uuid"
+)
+
+const (
+	operateTypeSeq = 1
```
[the remark this answers is not in the document]
Comment (giskook) [reply ethtxmanager/custodialassetshttp.go]: done

File: ethtxmanager/custodialassets.go (hunk 1 above)
Comment (giskook) [reply ethtxmanager/custodialassets.go]: done

File: ethtxmanager/ethtxmanager.go [hunk 4]
```diff
@@ -382,7 +382,14 @@ func (c *Client) monitorTx(ctx context.Context, mTx monitoredTx, logger *log.Log
 		logger.Debugf("unsigned tx %v created", tx.Hash().String())
 
 		// sign tx
-		signedTx, err = c.etherman.SignTx(ctx, mTx.from, tx)
+		if c.cfg.CustodialAssets.Enable {
+			signedTx, err = c.signTx(mTx, tx)
+			if err != nil {
+				c.halt(err)
```
Comment (giskook) [inline ethtxmanager/ethtxmanager.go:388]: add alarm

**Review by zjg555543 — APPROVED** [review]

**Pull request merged**